Added Some Functions

brodo97 opened the pull request:
Added some functions that I daily use. Note that I tested them only on a 8.1.0 italian server.

Added **celestial_queue** to get the remaining construction/research time on planets (untested on moons). This is useful if you don't want to spam request to check if you can build or research something.

Added **resources_settings** to set and get resources' settings. This can be used to set or simply check what settings you have on your planets' resources. For example, if you're low on deuterium you can set fusion_plant to 0 to avoid deuterium consumption.

Added **galaxy_debris** to get debris in a system or a slot. I found this particularly useful when you try to get moons or when someone has attacked you. If you manage to save the recyclers you can instantly send them to the debris field. Or you can recycle other fields (if you're fast enough)

Updated **spyreports**. This is so good if you want to farm inactive or poorly defended planets. Just automate the spying task and then read the spy reports. Now it returns a list of reports with detailed info like: resources, defences, fleet, buildings and research.

PiecePaperCode committed:
Idigate that pyogameNET is still in developement

PiecePaperCode committed:
pyogame is online now

brodo97 committed:
Update __init__.py

Added celestial_queue to get the remaining construction/research time on planets (untested on moons)
Added resources_settings to set and get resources' settings. Es: set fusion plant to 10%, metal mines to 100% etc..
Added galaxy_debris to get debris in a system or a slot
Updated spyreports, now it returns a detailed list of resources, defences, fleet, buildings and research

brodo97 committed:
Update constants.py

Added similar functions to ship_name. They convert the tech id/number to a human readable name. New spyreports function need these functions

brodo97 committed:
Update test.py

Updated for new functions

brodo97 committed:
Fixes

brodo97 committed:
Update README.md

brodo97 committed:
Fixed useless request

Now it avoids to make multiple requests if the user choose to just print the resources settings

PiecePaperCode commented on README.md:
```diff
@@ -208,6 +222,38 @@ res.crystal                             returns int
+            "crystal_mine": speed.min,
+            "fusion_plant": 0,
+            "solar_satellite": speed._50,
+        }
```
could you make the use of the constants instead of strings. you rely on the dev to memorise and write the correct string. if the string has a misspell it will fail only at runtime

brodo97 replied:
Oh, yeah! Actually, I could take a cue from the other functions like **send_fleet**, from the **fleet** parameter. Guess I need to move to an IDE

PiecePaperCode commented on ogame/__init__.py:
```diff
@@ -370,6 +399,47 @@ class Resources:
 
         return Resources
 
+    def resources_settings(self, id, settings={}):
```
settings cant be strings. missspels will fail this function. no ide support. this should be parameters that can be set optionally

PiecePaperCode reviewed (changes requested):
great work i relly like it alot :+1:

brodo97 committed:
Update __init__.py

Changed settings parameter of resources_settings from a dict of strings to a dict of buildings

brodo97 committed:
Update README.md

Some changes and fixes

brodo97 committed:
Fixed celestial_queue returned values

Fixed celestial_queue, now it returns only datetime values

brodo97 committed:
Update README.md

PiecePaperCode committed:
Overall Improvements. Make Test Pass exept Spyreports

PiecePaperCode commented:
I get this error on spyreports.
The spyreport didnt go through fully so data is missing failing the spyreport function
<pre>
ERROR: test_spyreports (ogame.test.UnittestOgame)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/Documents/repos/github/pyogame/ogame/test.py", line 195, in test_spyreports
    for report in self.empire.spyreports():
  File "/home/ubuntu/Documents/repos/github/pyogame/ogame/__init__.py", line 1144, in spyreports
    tech_id = int(r.match(li.find('img')['class'][0]).group(2))
TypeError: 'NoneType' object is not subscriptable
</pre>

PiecePaperCode commented:
You are using alot of set attribute for your classes.
would it be possible to use constructors instead of setting the class afterwards
![image](https://user-images.githubusercontent.com/40020346/127343625-cec71859-0109-4328-b429-99c58123eb84.png)
IDE's like pycharm fail to recognise your Data Structure. i need to look up documentation and typeerrors are not catched by the interpreter. it only fails at runtime.

brodo97 committed:
Fixes and IDE-Friendly mods

brodo97 commented:
Here we go. I installed PyCharm and tested it. Now it should be easier for the IDE to see returned data structure.
I added **api** parameter to each report.
Please let me know if there are other problems

PiecePaperCode commented on ogame/__init__.py:
```diff
@@ -370,6 +405,56 @@ class Resources:
 
         return Resources
 
+    def resources_settings(self, id, settings=None):
```
super clean impl

PiecePaperCode commented on ogame/__init__.py:
```diff
@@ -311,6 +311,41 @@ class Celestial:
+            build_time = int(build_time.group(1))
+            build_time = datetime.fromtimestamp(build_time)
+        shipyard_time = re.search(r'var restTimeship2 = ([0-9]+)', response)
+        if shipyard_time is None:
```
is None i think is redundant if shipyard: should be sufficiant
but bedder you return a 0 timestamp. not crashing bots from the enduser

PiecePaperCode commented on ogame/constants.py:
```diff
@@ -128,6 +164,24 @@ def is_research(research):
         else:
             return False
 
+    def research_name(res):
```
when i first coded this i didnt understand hasmaps.
but today i know them.
when i have the time i should convert all const to hasmaps that are implementet in if chains

PiecePaperCode commented:
```
test_Constants (ogame.test.UnittestOgame) ... ok
test_Events (ogame.test.UnittestOgame) ... ok
test_Vars (ogame.test.UnittestOgame) ... ok
test_ally (ogame.test.UnittestOgame) ... ok
test_build (ogame.test.UnittestOgame) ... ok
test_celestial (ogame.test.UnittestOgame) ... ok
test_celestial_coordinates (ogame.test.UnittestOgame) ... ok
test_celestial_queue (ogame.test.UnittestOgame) ... ok
test_collect_rubble_field (ogame.test.UnittestOgame) ... ok
test_defences (ogame.test.UnittestOgame) ... ok
test_facilities (ogame.test.UnittestOgame) ... ok
test_fleet (ogame.test.UnittestOgame) ... ok
test_galaxy (ogame.test.UnittestOgame) ... ok
test_galaxy_debris (ogame.test.UnittestOgame) ... ok
test_moon_facilities (ogame.test.UnittestOgame) ... ok
test_phalanx (ogame.test.UnittestOgame) ... ok
test_relogin (ogame.test.UnittestOgame) ... ok
test_research (ogame.test.UnittestOgame) ... ok
test_resources (ogame.test.UnittestOgame) ... ok
test_resources_settings (ogame.test.UnittestOgame) ... ok
test_return_fleet (ogame.test.UnittestOgame) ... ok
test_send_fleet (ogame.test.UnittestOgame) ... ok
test_send_message (ogame.test.UnittestOgame) ... ok
test_ships (ogame.test.UnittestOgame) ... ok
test_slot_celestial (ogame.test.UnittestOgame) ... ok
test_slot_fleet (ogame.test.UnittestOgame) ... ok
test_spyreports (ogame.test.UnittestOgame) ... ok
test_supply (ogame.test.UnittestOgame) ... ok

----------------------------------------------------------------------
Ran 28 tests in 256.329s

OK

Process finished with exit code 0
```
